app/app.rb

```diff
@@ -0,0 +1,78 @@
+ENV['RACK_ENV'] ||= "development"
+require 'sinatra/base'
+require 'sinatra/flash'
+require_relative 'data_mapper_setup'
+
+class Chitter < Sinatra::Base
+  use Rack::MethodOverride
+
+  enable :sessions
+    register Sinatra::Flash
+
+    set :session_secret, 'super secret'
+
+    helpers do
+     def user_name
+      @user_name ||= User.get(session[:user_id])
+     end
+    end
+
+    get '/users/new' do
+      erb :'users/new'
+    end
+
+    post '/users' do
+      user = User.new(params)
+      if user.save
+        session[:user_id] = user.id
+        redirect '/chat'
+      else
+        flash[:error] = user.errors.full_messages.join('. ')
+        flash[:email] = params[:email]
+        flash[:name] = params[:name]
+        flash[:username] = params[:username]
+        redirect '/users/new'
+      end
+    end
+
+      get '/chat' do
+        @peeps = Peep.all
+        erb :'links/chat'
+      end
+
+      get '/chat/add' do
+        erb :'links/add'
+      end
+
+      post '/chat' do
+        peep = Peep.create(message: params[:message], time: Time.new.asctime)
+          # params[:].split.each do |tag|
+          #   link.tags << Tag.create(tags: tag)
+          peep.save
+          redirect '/chat'
+       end
```

Comment:
Lint/BlockAlignment: `end` at 53, 7 is not aligned with `post '/chat' do` at 47, 6.